paddle/phi/kernels/onednn/slice_kernel.cc

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2022 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/phi/kernels/slice_kernel.h"
+
+#include "paddle/phi/backends/onednn/onednn_reuse.h"
+#include "paddle/phi/core/kernel_registry.h"
+
+namespace phi {
+
+template <typename T, typename Context>
+void SliceRawKernel(const Context& dev_ctx,
+                    const DenseTensor& x,
+                    const std::vector<int64_t>& axes,
+                    const IntArray& starts,
+                    const IntArray& ends,
+                    const std::vector<int64_t>& infer_flags,
+                    const std::vector<int64_t>& decrease_axis,
+                    DenseTensor* out) {
+  const auto& onednn_engine = dev_ctx.GetEngine();
+
+  auto x_vec_dims = vectorize(x.dims());
+
+  auto starts_vec = starts.GetData();
+  auto ends_vec = ends.GetData();
+
+  std::vector<int64_t> offsets(x_vec_dims.size(), 0);
+  std::vector<int64_t> slice_dims(x_vec_dims);
+
+  for (size_t i = 0; i < axes.size(); ++i) {
+    starts_vec[i] =
+        starts_vec[i] < 0 ? x_vec_dims[axes[i]] + starts_vec[i] : starts_vec[i];
+    ends_vec[i] = ends_vec[i] < 0 ? x_vec_dims[axes[i]] + ends_vec[i]
+                                  : std::min(ends_vec[i], x_vec_dims[axes[i]]);
+    offsets[axes[i]] = starts_vec[i];
+    slice_dims[axes[i]] =
+        std::max(static_cast<int64_t>(0), ends_vec[i] - starts_vec[i]);
+    ;
+  }
+
+  out->Resize(make_ddim(slice_dims));
+
+  // Note(0x45f): To support slice Tensors with shapes like [0, 0, 0].
+  if (!x.initialized()) {
+    out->mutable_data(x.place(), x.dtype());
```

Comment:
The mutable_data is deprecated in PHI and if you want to allocate memory in PHI, you can use "dev_ctx.template Alloc<T>(out)"